homeassistant/components/insteon/insteon_entity.py

```diff
@@ -0,0 +1,127 @@
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    DOMAIN,
+    INSTEON_ENTITIES,
+    SIGNAL_LOAD_ALDB,
+    SIGNAL_PRINT_ALDB,
+    STATE_NAME_LABEL_MAP,
+)
+from .utils import print_aldb_to_log
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class InsteonEntity(Entity):
+    """INSTEON abstract base entity."""
+
+    def __init__(self, device, state_key):
+        """Initialize the INSTEON binary sensor."""
+        self._insteon_device_state = device.states[state_key]
+        self._insteon_device = device
+        self._insteon_device.aldb.add_loaded_callback(self._aldb_loaded)
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def address(self):
+        """Return the address of the node."""
+        return self._insteon_device.address.human
+
+    @property
+    def group(self):
+        """Return the INSTEON group that the entity responds to."""
+        return self._insteon_device_state.group
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        if self._insteon_device_state.group == 0x01:
+            uid = self._insteon_device.id
+        else:
+            uid = "{:s}_{:d}".format(
+                self._insteon_device.id, self._insteon_device_state.group
+            )
+        return uid
+
+    @property
+    def name(self):
+        """Return the name of the node (used for Entity_ID)."""
+        # Set a base description
+        description = self._insteon_device.description
+        if self._insteon_device.description is None:
+            description = "Unknown Device"
+
+        # Get an extension label if there is one
+        extension = self._get_label()
+        if extension:
+            extension = f" {extension}"
+        name = "{:s} {:s}{:s}".format(
+            description, self._insteon_device.address.human, extension
+        )
+        return name
+
+    @property
+    def device_state_attributes(self):
+        """Provide attributes for display on device card."""
+        attributes = {"INSTEON Address": self.address, "INSTEON Group": self.group}
+        return attributes
+
+    @callback
+    def async_entity_update(self, deviceid, group, val):
+        """Receive notification from transport that new data exists."""
+        _LOGGER.debug(
+            "Received update for device %s group %d value %s",
+            deviceid.human,
+            group,
+            val,
+        )
+        self.async_schedule_update_ha_state()
+
+    async def async_added_to_hass(self):
+        """Register INSTEON update events."""
+        _LOGGER.debug(
+            "Tracking updates for device %s group %d statename %s",
+            self.address,
+            self.group,
+            self._insteon_device_state.name,
+        )
+        self._insteon_device_state.register_updates(self.async_entity_update)
+        self.hass.data[DOMAIN][INSTEON_ENTITIES][self.entity_id] = self
```

Comment:
Don't store the entity instance. Just store the entity_id. Entities are a platform concern.